lib/contact_screen.dart

```diff
@@ -0,0 +1,317 @@
+                  color: Theme.of(context).textSelectionTheme.selectionColor),
+              onPressed: () {
+                _valueNotifier.value = !_valueNotifier.value;
+              },
+            )),
+        title: titleSearchWidget(),
+        actions: <Widget>[
+          IconButton(
+            icon: Icon(Icons.close,
+                color: Theme.of(context).textSelectionTheme.selectionColor),
+            onPressed: () {
+              _valueNotifier.value = !_valueNotifier.value;
+            },
+          ),
+        ]);
+  }
+
+  void searchFromContactList() {
+    if (_searchController.text.isNotEmpty) {
+      if (_filterOn == "name") {
+        final list = _contactsList.where((element) {
+          return element.name
+              .toLowerCase()
+              .contains(_searchController.text.toLowerCase());
+        }).toList();
+        _searchNotifier.updateSearchValue(list);
+      } else if (_filterOn == "phoneNumber") {
+        final list = _contactsList.where((element) {
+          if (element.phoneNumber.isNotEmpty) {
+            return element.phoneNumber
+                .toLowerCase()
+                .contains(_searchController.text.toLowerCase());
+          } else {
+            return false;
+          }
+        }).toList();
+        _searchNotifier.updateSearchValue(list);
+      } else if (_filterOn == "kilvishId") {
+        final list = _contactsList.where((element) {
+          if (element.kilvishId != null) {
+            return element.kilvishId!
+                .toLowerCase()
+                .contains(_searchController.text.toLowerCase());
+          } else {
+            return false;
+          }
+        }).toList();
+        _searchNotifier.updateSearchValue(list);
+      }
+    } else {
+      _searchNotifier.updateSearchValue(_contactsList);
+    }
+  }
+
+  /// Showing title widget for app bar Search title
+  Widget titleSearchWidget() {
+    return TextField(
+      controller: _searchController,
+      onChanged: (value) {
+        searchFromContactList();
+      },
+      decoration: InputDecoration(
+          prefixIcon: Icon(Icons.search,
+              color: Theme.of(context).textSelectionTheme.selectionColor),
+          hintText: _hintText,
+          suffixIcon: PopupMenuButton<String>(
+            icon: const Icon(Icons.filter_list),
+            onSelected: (String result) {
+              _filterOn = result;
+              searchFromContactList();
+            },
+            itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
+              const PopupMenuItem<String>(
+                value: 'name',
+                child: Text('Name'),
+              ),
+              const PopupMenuItem<String>(
+                value: 'phoneNumber',
+                child: Text('Phone Number'),
+              ),
+              const PopupMenuItem<String>(
+                value: 'kilvishId',
+                child: Text('Kilvish Id'),
+              ),
+            ],
```

Comment:
Lets not give user option to decide which field to search .. we should just search ourselves.